sonar-text-plugin/src/main/java/org/sonar/plugins/text/checks/AbstractInputStreamCheck.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.text.checks;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.sonar.plugins.text.api.CheckContext;
+import org.sonar.plugins.text.api.CommonCheck;
+import org.sonar.plugins.text.api.InitContext;
+
+/**
+ * Abstract Check to commonly catch possible IOExceptions for InputFiles
+ */
+public abstract class AbstractInputStreamCheck implements CommonCheck {
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register((ctx, inputFile) -> {
+      try {
+        analyzeStream(ctx, inputFile.inputStream());
```

Comment:
The doc of `InputFile#inputStream()` says that the caller should close the stream ([link](https://github.com/SonarSource/sonarqube/blob/d91ea92fe22cf1b6b52a8efba122daf8e7e58282/sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/InputFile.java#L121-L128)). Let's do this here. We can change this into a try-with-resources (`try (InputStream = inputFile.inputStream()) {...}`)